### PR TITLE
Merge surface jumps 4

### PR DESCRIPTION
**Pull request by drwells**
### IBAMR Pull Request Checklist
- [x] Does the test suite pass?
- [x] Was clang-format run?
- [x] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [x] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [x] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

**Comment by drwells** [comment]
Like the others the only relevant commit is the last one.

File: include/ibamr/IBFESurfaceMethod.h [hunk 1]
```diff
@@ -564,12 +564,8 @@ class IBFESurfaceMethod : public IBStrategy
 
     SAMRAI::hier::IntVector<NDIM> d_ghosts = 0;
 
-    std::vector<libMesh::System*> d_X_systems, d_U_systems, d_U_n_systems, d_U_t_systems, d_F_systems, d_DP_systems;
+    std::vector<libMesh::System*> d_U_systems, d_DP_systems;
```
Comment (boyceg) [inline include/ibamr/IBFESurfaceMethod.h:567]: 👍

File: include/ibamr/IBFESurfaceMethod.h [hunk 2]
```diff
@@ -581,6 +577,44 @@ class IBFESurfaceMethod : public IBStrategy
...
+     * Object managing access to libMesh system vectors for the velocity system.
+     *
+     * @note Unlike IBAMR::IBFEMethod, this class does not inherit from
+     * FEMechanicsBase and therefore both normal and IB vectors are handled by
+     * d_U_vecs.
+     */
+    std::unique_ptr<IBTK::LibMeshSystemIBVectors> d_U_vecs;
+
+    /*!
+     * Object managing access to libMesh system vectors for the normal velocity
```
Comment (boyceg) [inline include/ibamr/IBFESurfaceMethod.h:590]: Here's a question --- do we still need all of those different systems, or could we just have `U`, `U_n`, and `U_t` system vectors?

Comment (boyceg) [reply include/ibamr/IBFESurfaceMethod.h:590]: We had different systems originally because my vector management "system" was terrible.

Comment (drwells) [reply include/ibamr/IBFESurfaceMethod.h:590]: I think at some point we decided that it was useful to visualize these things, which means they really do need their own `System`.

**Review by boyceg — APPROVED** [review]

File: src/IB/IBFESurfaceMethod.cpp [hunk 3]
```diff
@@ -303,60 +303,18 @@ IBFESurfaceMethod::preprocessIntegrateData(double current_time, double new_time,
...
-
-    d_U_t_systems.resize(d_num_parts);
-    d_U_t_current_vecs.resize(d_num_parts);
-    d_U_t_new_vecs.resize(d_num_parts);
-    d_U_t_half_vecs.resize(d_num_parts);
-
-    d_F_systems.resize(d_num_parts);
-    d_F_half_vecs.resize(d_num_parts);
-    d_F_IB_ghost_vecs.resize(d_num_parts);
+    d_U_vecs->copy("solution", { "current", "new", "half" });
```
Comment (boyceg) [inline src/IB/IBFESurfaceMethod.cpp:306]: This pattern is getting pretty common --- I wonder if it is worth batching it up, e.g.,
```
batch_vec_copy( {d_X_vecs, d_U_vecs, d_U_n_vecs, d_U_t_vecs}, "solution", {"current", "new", "half"});
```

File: src/IB/IBFESurfaceMethod.cpp [hunk 4]
```diff
@@ -1448,6 +1317,10 @@ void IBFESurfaceMethod::endDataRedistribution(Pointer<PatchHierarchy<NDIM> > /*h
     }
 
     d_X_vecs->reinit();
+    d_U_vecs->reinit();
```
Comment (boyceg) [inline src/IB/IBFESurfaceMethod.cpp:1320]: Similarly here --- some way of batching together calls to `reinit()`.

**Review by boyceg — APPROVED** [review]

**Comment by drwells** [comment]
Do you want to just merge this one and close the others? I split it up to make it easier to review but its not a big deal either way.

If so - I'll increment the restart version number and then press the big green button.

**Comment by boyceg** [comment]
I don't care how it gets merged as long as it all eventually gets sorted out.

File: src/IB/IBFESurfaceMethod.cpp (hunk 3 above)
Comment (drwells) [reply src/IB/IBFESurfaceMethod.cpp:306]: Maybe - the IIM project adds a lot more systems, though, so I'd rather get more of that merged first before we add more utility functions.

**Pull request merged**